sdk/keyvault/keyvault-certificates/src/identifier.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { parseKeyvaultIdentifier } from "./core/utils";
+
+/**
+ * Represents a KeyVault identifier and its parsed contents.
+ */
+export interface ParsedKeyVaultCertificatesIdentifier {
+  /**
+   * The type of resource under KeyVault that this identifier is referring to.
+   */
+  collection: "certificates";
+
+  /**
+   * The originally received identifier.
+   */
+  id: string;
+
+  /**
+   * The KeyVault Key unique identifier (an URl).
+   */
+  vaultUrl: string;
+
+  /**
+   * The version of KeyVault Key. May be undefined.
+   */
+  version?: string;
+
+  /**
+   * The name of the KeyVault Key.
+   */
+  name: string;
+}
+
+/**
+ * Parser of the KeyVaultIdentifier for the KeyVault Keys Client.
+ */
+export class KeyVaultCertificatesIdentifier implements ParsedKeyVaultCertificatesIdentifier {
+  /**
+   * The type of resource under KeyVault that this identifier is referring to.
+   */
+  collection: "certificates";
+
+  /**
+   * The originally received identifier.
+   */
+  public id: string;
+
+  /**
+   * The KeyVault Key unique identifier (an URl).
+   */
+  public vaultUrl: string;
+
+  /**
+   * The version of KeyVault Key. May be undefined.
+   */
+  public version?: string;
+
+  /**
+   * The name of the KeyVault Key.
+   */
+  public name: string;
+
+  constructor(url: string) {
```

Comment:
FWIW, .NET won't be doing this in construction. Constructors shouldn't generally throw on bad data (but should on developer issues, like null required arguments, for example). .NET's pattern is typically a `TryParse` or optionally `Parse` (the latter is older, but more and more is moving to *only* the former). Doesn't JS use a similar pattern for parsing into objects?

Reply:
I'll check the guidelines.

Reply:
I can't see anything about throwing exceptions on constructors. I will ask.

Reply:
Asked @xirzec and he said this was ok (though I did ask with an abstract example, Jeff! Here's the real scenario!)